src/backend/Device.cpp

```diff
@@ -98,8 +98,23 @@ namespace backend {
     BindGroupBuilder* DeviceBase::CreateBindGroupBuilder() {
         return new BindGroupBuilder(this);
     }
-    BindGroupLayoutBuilder* DeviceBase::CreateBindGroupLayoutBuilder() {
-        return new BindGroupLayoutBuilder(this);
+    BindGroupLayoutBase* DeviceBase::CreateBindGroupLayout(
```

Comment:
BindGroups are a bit different from other objects because they need to be deduplicated to allow comparison by pointer (helps make the check that BGs match the PL fast). The validation code part of the code would stay the same, then a BindGroupLayoutBase would be created with blueprint=true and finally GetOrCreateBindGroupLayout called.

Reply:
In the future we should add a validation test that a if BGL1 == BGL2 then a BG of layout BGL1 can be used with a PL slot with layout BGL2.

Reply:
Oops, didn't realize I broke this. Fixed (and added test)